packages/web-ui/src/lab/Button/Button.tsx

```diff
@@ -0,0 +1,331 @@
+      '--button-foreground-color-disabled': 'var(--button-ghost-foreground-color-disabled)',
+      ':not(:hover,:active,:disabled)': {
+        '--button-icon-color': 'var(--button-ghost-icon-color)',
+      },
+    },
+    [classSelectors.variant.outline]: {
+      '--button-background-color': 'transparent',
+      '--button-background-color-disabled': 'transparent',
+      '--button-foreground-color': 'var(--button-outline-foreground-color)',
+      '--button-background-color-hover': 'var(--button-outline-background-color-hover)',
+      '--button-background-color-active': 'var(--button-outline-background-color-active)',
+      '--button-foreground-color-disabled': 'var(--button-outline-foreground-color-disabled)',
+      '--button-border-color-disabled': 'var(--button-outline-border-color-disabled)',
+      '--button-outline-border': 'inset 0 0 0 2px var(--button-outline-border-color)',
+      boxShadow: 'var(--button-outline-border)',
+      ':not(:hover,:active,:disabled)': {
+        '--button-icon-color': 'var(--button-outline-icon-color)',
+      },
+    },
+    [classSelectors.size.large]: { ...sizeStyles.large },
+    [classSelectors.size.small]: { ...sizeStyles.small },
+    [mediaQueries.tablet]: {
+      [classSelectors.size.tablet.large]: { ...sizeStyles.large },
+      [classSelectors.size.tablet.small]: { ...sizeStyles.small },
+    },
+    [mediaQueries.desktop]: {
+      [classSelectors.size.desktop.large]: { ...sizeStyles.large },
+      [classSelectors.size.desktop.small]: { ...sizeStyles.small },
+    },
+    [mediaQueries.wide]: {
+      [classSelectors.size.wide.large]: { ...sizeStyles.large },
+      [classSelectors.size.wide.small]: { ...sizeStyles.small },
+    },
+    '&:where(:focus-visible)': {
+      boxShadow: 'var(--button-focus-outline)',
+      '--button-background-color': 'var(--button-background-color-hover)',
+      [classSelectors.variant.outline]: {
+        boxShadow: 'var(--button-outline-border), var(--button-focus-outline)',
+      },
+    },
+    '@media (hover: hover)': {
+      cursor: 'default', // so the cursor doesn't change on hover when using a `a` element with `asChild`
+      '&:where(:hover)': {
+        '--button-background-color': 'var(--button-background-color-hover)',
+      },
+    },
+    '&:where(:active)': {
+      '--button-background-color': 'var(--button-background-color-active)',
+      '--button-icon-color': 'var(--button-icon-color-active)',
+    },
+    [dataAttributes.disabled]: {
+      cursor: 'not-allowed',
+      '--button-foreground-color': 'var(--button-foreground-color-disabled)',
+      '--button-background-color': 'var(--button-background-color-disabled)',
+      '--button-border-color': 'var(--button-border-color-disabled)',
+      '--button-icon-color': 'var(--button-icon-color-disabled)',
+      [classSelectors.variant.outline]: {
+        '--button-outline-border-color': 'var(--button-outline-border-color-disabled)',
+      },
+    },
+  };
```

Comment:
this is a big chunk of styling. Maybe it would be better placed in a separate file?

Reply:
yeah, I did consider this before, I ended up with a weird circular dependency, but will have another look if there's a logical way of splitting this ➕